sources/packages/api/src/services/education-program/education-program.service.ts

```diff
@@ -483,9 +515,105 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
         "programs.intlExchangeProgramEligibility",
         "programs.programDeclaration",
         "institution.id",
+        "institution.legalOperatingName",
+        "programs.submittedOn",
+        "submittedBy.firstName",
+        "submittedBy.lastName",
+        "statusUpdatedBy.firstName",
+        "statusUpdatedBy.lastName",
+        "programs.statusUpdatedOn",
+        "programs.effectiveEndDate",
       ])
+      .leftJoin("programs.submittedBy", "submittedBy")
+      .leftJoin("programs.statusUpdatedBy", "statusUpdatedBy")
       .innerJoin("programs.institution", "institution")
       .where("programs.id = :id", { id: programId })
       .getOne();
   }
+
+  /**
+   * Approve a pending program.
+   * @param programId program id.
+   * @param userId user id.
+   * @param payload ApproveProgram
+   * @param returns EducationProgram
+   */
+  async approveEducationProgram(
+    programId: number,
+    userId: number,
+    payload: ApproveProgram,
+  ): Promise<EducationProgram> {
+    const program = await this.repo.findOne(
+      {
+        id: programId,
+        approvalStatus: ApprovalStatus.pending,
+      },
+      {
+        relations: ["programNote", "statusUpdatedBy"],
```

Comment:
I believe that there is no need to load the relations here since they will be entirely replaced.